src/main/java/org/wso2/carbon/connector/pcml/AS400CallProgram.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.connector.pcml;
+
+import com.ibm.as400.access.AS400;
+import com.ibm.as400.access.AS400Message;
+import com.ibm.as400.access.ExtendedIllegalArgumentException;
+import com.ibm.as400.data.PcmlException;
+import com.ibm.as400.data.ProgramCallDocument;
+import com.ibm.as400.data.XmlException;
+import javax.xml.stream.XMLStreamException;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.util.AXIOMUtil;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.SynapseLog;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * A connector component that calls an AS400 program using PCML.
+ */
+public class AS400CallProgram extends AbstractConnector {
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     *     Calls a program in the AS400 server using PCML. The input parameters are taken through the soap body of the
+     *     message context.
+     * </p>
+     */
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        SynapseLog log = getLog(messageContext);
+        AS400 as400 = null;
+        try {
+            Object as400InstanceProperty = messageContext.getProperty(AS400Constants.AS400_INSTANCE);
+            if (null != as400InstanceProperty) {
+                as400 = (AS400) as400InstanceProperty;
+            } else {
+                throw new AS400PCMLConnectorException("Unable to find an AS400 instance to call program. Use the " +
```

Comment:
Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/files#diff-662ab4a619a4739c3828ff890f9fe024R65